src/snowflake/snowpark/modin/plugin/_internal/get_dummies_utils.py

```diff
@@ -0,0 +1,311 @@
+    """
+    Helper function for get dummies to perform a single pivot on the encoded column.
+    Args:
+        internal_frame: The original internal frame to perform pivot on.
+            Note: the input internal frame must have a row position column and dummy lit(1) column
+                as the last data column
+        prefix: String to append to newly generated column names after pivot
+        prefix_sep: The separator used between the prefix and new column names
+        pivot_column_snowflake_quoted_identifier: The encoded column, which is the column to pivot on
+        columns_to_keep_snowflake_quoted_identifiers: The snowflake quoted identifier in the
+            internal_frame to keep as the data column of final result internal frame.
+        columns_to_keep_pandas_labels: The pandas label in the internal_frame to keep as the
+            data_column of final result internal frame.
+
+        Note: columns_to_keep_snowflake_quoted_identifiers must be the same length as columns_to_keep_pandas_labels
+    Returns:
+        InternalFrame: An InternalFrame whose data columns are the pivoted result columns + the columns_to_keep,
+            and the row position column of the original internal_frame as index column.
+
+    Example:
+        With the following DataFrame, where lit_one is the dummy lit one column:
+
+             A  C lit_one
+          0  a  1   1
+          1  b  2   1
+          2  a  3   1
+
+        the result of calling single_get_dummies_pivot with ['"C"]' as column_to_keep,
+        "A" as prefix, and "_" as prefix_sep will be the following:
+
+            C  A_a  A_b
+         0  1    1    0
+         1  2    0    1
+         2  3    1    0
+    """
+
+    # get the row position column and dummy lit one column
+    assert internal_frame.row_position_snowflake_quoted_identifier is not None
+    row_position_snowflake_quoted_identifier = (
+        internal_frame.row_position_snowflake_quoted_identifier
+    )
+    dummy_column_snowflake_quoted_identifier = (
+        internal_frame.data_column_snowflake_quoted_identifiers[-1]
+    )
+    # for the dataframe passed for pivot, we keep the row position column, dummy lit one column,
+    # pivot column, and the specified columns_to_keep
+    columns_snowflake_quoted_identifier = [
+        row_position_snowflake_quoted_identifier,
+        pivot_column_snowflake_quoted_identifier,
+        dummy_column_snowflake_quoted_identifier,
+    ] + columns_to_keep_snowflake_quoted_identifiers
+    ordered_dataframe = internal_frame.ordered_dataframe.select(
+        columns_snowflake_quoted_identifier
+    )
+    # Perform pivot on the pivot column with dummy lit one column as value column.
+    # With the above example, the result of pivot will be:
+    #
+    #    C    a    b
+    # 0  1    1    0
+    # 1  2    0    1
+    # 2  3    1    0
+    pivoted_ordered_dataframe = ordered_dataframe.pivot(
+        col(str(pivot_column_snowflake_quoted_identifier)),
```

Comment:
@sfc-gh-evandenberg It seems that we might have paying the dynamic subquery execution overhead multiple times (during describing and actual execution),  i expect the query result caching could help to eliminate the second subquery elimination. Can we follow up on taking a look at the performance for this part at server side?

Reply:
Can you you create a jira to investigate here?  If this relates to the query result caching then can also ask in the SQL help slack channel as they can likely provide some pointers.  It's not clear this is pivot so much as caching of any duplicate queries.